lib/widgets/medication_details.dart

```diff
@@ -0,0 +1,670 @@
+import 'package:connect2bahmni/domain/models/bahmni_drug_order.dart';
+import 'package:connect2bahmni/services/concept_dictionary.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class MedicationDetails extends StatefulWidget {
+  final BahmniDrugOrder medication;
+
+  const MedicationDetails({Key? key, required this.medication})
+      : super(key: key);
+
+  @override
+  State<MedicationDetails> createState() => _MedicationDetailsState();
+}
+
+class _MedicationDetailsState extends State<MedicationDetails> {
+  final _formKey = GlobalKey<FormState>();
+  String? _selectedUnitType;
+  String? _selectedRouteType;
+  String? _selectedFrequencyType;
+  String? _selectedDurationType;
+  String? _selectedDosingInstructions;
+  String lblTitle = " ";
+  late BahmniDrugOrder _medication;
+  late final Map<dynamic, dynamic>? _dosageInstructions;
+  final TextEditingController _notesController = TextEditingController();
+  final TextEditingController _dateController = TextEditingController();
+  final TextEditingController _doseController = TextEditingController();
+  final TextEditingController _durationController = TextEditingController();
+  final TextEditingController _t1Controller = TextEditingController();
+  final TextEditingController _t2Controller = TextEditingController();
+  final TextEditingController _t3Controller = TextEditingController();
+  static const lblAddMedication = "Add Medication";
+  static const lblEnterMedicationNote = "Enter note for the medication";
+  final ValueNotifier<bool> _typeChangeAllowed = ValueNotifier<bool>(true);
+  double totalQuantity = 0.0;
+  late Future<Map?> _dataFuture;
+  bool toggle=false;
+  double s1=0;
```

Comment:
better name for variables please? morningDose, afternoonDose, eveningDose etc.

Reply:
Altered